pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -180,24 +180,43 @@ func (r *ReconcileDeadmansSnitchIntegration) Reconcile(request reconcile.Request
 	}
 
 	for _, clusterdeployment := range allClusterDeployments.Items {
-		if clusterdeployment.DeletionTimestamp != nil {
+
+		// Check if the cluster matches the requirements for needing DMS setup
+		clusterMatched := false
+		for _, matchingClusterDeployment := range matchingClusterDeployments.Items {
+			if clusterdeployment.UID == matchingClusterDeployment.UID {
+				clusterMatched = true
+				break
+			}
+		}
+
+		if !clusterMatched {
+			// The cluster does not match the criteria for needing DMS setup
 			if utils.HasFinalizer(&clusterdeployment, deadMansSnitchFinalizer) {
+				// The cluster has an existing DMS setup, so remove it
 				err = r.deleteDMSClusterDeployment(dmsi, &clusterdeployment, dmsc)
 				if err != nil {
 					return reconcile.Result{}, err
 				}
 			}
+			continue
 		}
-	}
 
-	for _, clusterdeployment := range matchingClusterDeployments.Items {
+		// The cluster either has an existing (or requires a new) DMS setup
+
 		if clusterdeployment.DeletionTimestamp != nil {
-			// no action required, as this should be handled by the all function above
+			// The clusterdeployment is being deleted, so remove DMS
+			if utils.HasFinalizer(&clusterdeployment, deadMansSnitchFinalizer) {
+				err = r.deleteDMSClusterDeployment(dmsi, &clusterdeployment, dmsc)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+			}
 			continue
 		}
```

Comment:
Could collapse these into a single block, 
```
if !clusterMatched || clusterdeployment.DeletionTimestamp != nil {
```

Reply:
Thanks, that does indeed make things simpler - I've done this now.